letsencrypt/auth_handler.py

```diff
@@ -153,13 +153,18 @@ def _send_responses(self, achalls, resps, chall_update):
         """
         active_achalls = []
         for achall, resp in itertools.izip(achalls, resps):
+            # XXX: make sure that all achalls, including those
+            # corresponding to None or False returned from
+            # Authenticator are removed from the queue and thus avoid
+            # infinite loop
+            active_achalls.append(achall)
+
             # Don't send challenges for None and False authenticator responses
-            if resp:
+            if resp is not None and resp:
```

Comment:
I am pretty sure it used to be that way and it was switched... either way it doesn't really matter.